src/core/lib/surface/completion_queue.cc

```diff
@@ -448,14 +487,14 @@ grpc_completion_queue* grpc_completion_queue_create_internal(
   gpr_ref_init(&cq->owning_refs, 2);
 
   poller_vtable->init(POLLSET_FROM_CQ(cq), &cq->mu);
-  vtable->init(DATA_FROM_CQ(cq));
+  vtable->init(DATA_FROM_CQ(cq), shutdown_callback);
 
   GRPC_CLOSURE_INIT(&cq->pollset_shutdown_done, on_pollset_shutdown_done, cq,
                     grpc_schedule_on_exec_ctx);
   return cq;
 }
 
-static void cq_init_next(void* ptr) {
+static void cq_init_next(void* ptr, grpc_core::CQCallbackInterface*) {
```

Comment:
I will change this. It's actually not a good idea since it obscures the intent of the (unused) parameter. I need to make sure that it doesn't cause a compiler warning, though.

Reply:
I can make this stronger by asserting that shutdown_callback is nullptr in the next and pluck CQs, thus avoiding any chance of a compiler warning and also actually making sure that our expected invariant is held.